mne/time_frequency/_stockwell.py

```diff
@@ -55,51 +54,45 @@ def _precompute_st_windows(M, start_f, stop_f, sfreq, width):
                       np.exp(-0.5 * (1. / k ** 2.) * (f ** 2.) * tw ** 2.))
         window /= window.sum()  # normalisation
         windows[i_f] = fftpack.fft(window)
-    return M, f_range, windows
+    return windows
 
 
-def _st(x, M, f_range, windows):
-    """Implementation based on Matlab code by Ali Moukadem"""
-    if x.ndim == 1:
-        x = x[np.newaxis, :]
+def _st(x, start_f, windows):
+    """Implementation based on Ali Moukadem Matlab code (only used in tests)"""
+    n_samp = x.shape[-1]
+    ST = np.empty(x.shape[:-1] + (len(windows), n_samp), dtype=np.complex)
+    # do the work
     Fx = fftpack.fft(x)
-    XF = np.c_[Fx, Fx]
-    ST = np.empty((x.shape[0], len(f_range), M), dtype=np.complex)
-    for i_f, (window, f) in enumerate(zip(windows, f_range)):
-        for i in range(len(x)):
-            ST[i, i_f, :] = fftpack.ifft(XF[i, f:f + M] * window)
-    if ST.shape[0] == 1:
-        ST = ST[0]
+    XF = np.concatenate([Fx, Fx], axis=-1)
```

Comment:
My impression was that `np.c_` was not supposed to be used in production code... not sure if it's faster or slower